extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1482,12 +1592,82 @@ export class Project implements ISqlProject {
 		const firstPropertyElement = propertyElements[0];
 		if (firstPropertyElement.childNodes.length !== 1) {
 			// Property items are expected to have simple string content
-			throw new Error(constants.invalidProjectPropertyValue(propertyName));
+			throw new Error(constants.invalidProjectPropertyValueInSqlProj(propertyName));
 		}
 
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Sets the value of project property, overrides value if already present in project file.
+	 * @param propertyName Name of property
+	 * @param propertyValue Value of property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string): Promise<void> {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		let propertyGroup = propertyGroups.length > 0 ? propertyGroups[0] : null;
+		if (propertyGroup === null) {
+			propertyGroup = this.projFileXmlDoc.createElement(constants.PropertyGroup);
+			this.projFileXmlDoc.documentElement?.appendChild(propertyGroup);
+		}
+
+		const propertyElements = propertyGroup.getElementsByTagName(propertyName);
+		let propertyElement: Element | undefined;
+		if (propertyElements.length === 0) {
+			propertyElement = this.addProjectPropertyTag(propertyName);
+			propertyElement?.appendChild(this.projFileXmlDoc.createTextNode(propertyValue));
```

Comment:
Just checking - does it make it appear _inline_, i.e.
```xml
<MyProperty>value</MyProperty>
```
or formatted
```xml
<MyProperty>
   value
</MyProperty>
```
?

While both should work, we probably want inline, as that's typically how msbuild properties are defined.

Also, when doing any processing of the property, do we trim whitespaces and line breaks before/after the value? Given that it can be defined in the formatted form, it should not affect the values we extract from the property.